lib/javascript/BasicEvaluatedExpression.js

```diff
@@ -402,16 +402,27 @@ class BasicEvaluatedExpression {
 		return this;
 	}
 
-	setTruthy() {
-		this.falsy = false;
-		this.truthy = true;
-		this.nullish = false;
+	/**
+	 * @param {boolean} state state
+	 * @returns {BasicEvaluatedExpression} this
+	 */
+	setTruthy(state) {
+		this.falsy = !state;
+		this.truthy = state;
+
+		if (state) {
+			this.nullish = false;
+		}
 		return this;
 	}
 
-	setFalsy() {
-		this.falsy = true;
-		this.truthy = false;
+	/**
+	 * @param {boolean} state state
+	 * @returns {BasicEvaluatedExpression} this
+	 */
+	setFalsy(state) {
+		this.falsy = state;
+		this.truthy = !state;
 		return this;
 	}
```

Comment:
This makes these method a little bit confusing. And this might also be considered as breaking change.
In addition to that `.setFalsy(false)` makes the expression truthy which is probably not intended.

Better keep `setTruthy` and `setFalsy` and call them conditionally instead.